content/articles/garbage-collection-in-c#-dotnet-framework/index.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /garbage-collection-in-c#-dotnet-framework/
+title: Garbage Collection in C#.NET Framework
+description: In this article, the reader will be walked through garbage collection in C# Dotnet framework.
+author: elijah-maina
+date: 2021-10-12T00:00:00-15:26
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url:  /engineering-education/garbage-collection-in-c#-dotnet-framework/hero.jpg
+    alt: Garbage collection in c# dotnet framework image
+---
+
+Garbage collection in the C#.Net framework is used to manage Memory automatically. In runtime, heap storage space can be allocated to an object class. If the software has completed all the activities related to the object, the memory location will be a waste since it will not be usable.
+ <!--more-->
+### Introduction
+Garbage collection is a critical approach in the .Net framework for releasing unused handled code objects from Memory and allowing the process to run faster. In this tutorial, we will go over the fundamentals of garbage collection.
+
+### Table of contents
+- [Overview on Garbage Collection](#overview-on-garbage-collection)
+- [Garbage Collections Different phases](#garbage-collections-different-phases)
+- [Garbage Collection Heap generation](#garbage-collection-heap-generation)
+- [Methods in Garbage Collection Class](#methods-in-garbage-collection-class)
+- [Stand out advantages of Garbage Collection](#stand-out-advantages-of-garbage-collection)
+
+### Overview on garbage collection
+In a .Net model, memory management is usually automated through the help of Garbage collection. A stack storage space is assigned only when a runtime class object is generated. Whenever the software completes all the activities connected with the thing, the Memory linked to it will not be helpful as it will not be utilized.
+
+Memory space that is not required is freed up with the help of garbage collection, which makes it worthwhile to use. Inbuilt scripting engine, the `Optimization Engine` found in the managed heap, is always used in garbage collection.
+When at least one of several conditions is met, garbage collection occurs. 

```

Comment:
Fix this line break and others like it. Preferably connected to the previous paragraph.